### PR TITLE
Update Mono Pong

### PR DESCRIPTION
**Pull request by Insane96**
## Scene
The scene had some problems:
All the collision node were twice as small as they should have. Not sure if it's due to an update in the engine.
image
So the collisions are now bigger (2x exactly) to cover the whole sprite, and I've changed the ball collision shape to a sphere.

## Ceiling and Floor
The ceiling was just transposing the angle of the ball by 90°, I've changed it to actually bounce back.

## Paddle
The paddles have been made faster in movement (+50%) and now bounce the ball back based off the position of the collision.

## Ball
Ball now speeds up faster and picks a new random direction and angle when the game starts or resets.

### CODE REVIEW

File: mono/pong/Logic/Paddle.cs
Comment (paulloz) [inline mono/pong/Logic/Paddle.cs:1]: Use 4 spaces (no tabs) to indent[^1].

[^1]: https://docs.godotengine.org/en/stable/tutorials/scripting/c_sharp/c_sharp_style_guide.html#formatting

**Review by paulloz — CHANGES_REQUESTED** [review]

**Comment by Insane96** [comment]
Applied all request changes

File: mono/pong/Logic/Ball.cs [hunk 1]
```diff
@@ -1,29 +1,37 @@
+using System;
 using Godot;
 
 public partial class Ball : Area2D
 {
+    private readonly Random _random = new();
```
Comment (aaronfranke) [inline mono/pong/Logic/Ball.cs:6]: Target-typed `new` is only available in C# 9 or later. For the sake of keeping the demo's code compatible with older C# versions, we should not use it.

Keep in mind that even if Godot supports C# 9 or later, there are older Godot versions that do not. There is value in keeping the code consistent between all versions of the demo, even with the 3.x versions. That isn't to say that we don't want to improve things over time, just avoid unnecessary changes that don't affect behavior but do negatively affect compatibility.

**Review by aaronfranke — CHANGES_REQUESTED** [review]
None of these changes can be considered or accepted unless the same changes are made to the GDScript version of the demo.

File: mono/pong/Logic/Ball.cs [hunk 2]
```diff
@@ -1,29 +1,37 @@
...
-    public Vector2 direction = Vector2.Left;
+    public Vector2 Direction { get; set; }
 
     private Vector2 _initialPos;
     private double _speed = DefaultSpeed;
 
     public override void _Ready()
     {
         _initialPos = Position;
+        Direction = this.NewRandomDirection();
```
Comment (AThousandShips) [inline mono/pong/Logic/Ball.cs:17]: Are all the added `this` really necessary? What are they disambiguating?

Comment (paulloz) [reply mono/pong/Logic/Ball.cs:17]: Thanks, I missed those when I reviewed.

**Comment by paulloz** [comment]
> None of these changes can be considered or accepted unless the same changes are made to the GDScript version of the demo.

The purely corrective parts (area sizes in the `.tscn`, and the typo fixed in `Paddle.cs`) could. But otherwise, yes.

**Comment by AThousandShips** [comment]
Even those changes to the shapes don't match the gdscript version so they should match

**Pull request closed without merging**